libs/labelbox/src/labelbox/schema/api_key.py

```diff
@@ -0,0 +1,395 @@
+
+        Returns:
+            Dict[str, str]: Dictionary containing the created API key details including id and jwt
+
+        Raises:
+            ValueError: If invalid parameters are provided
+            LabelboxError: If the API request fails or there are permission issues
+        """
+        if not name or not isinstance(name, str):
+            raise ValueError("name must be a non-empty string")
+
+        user_email = user.email if hasattr(user, "email") else user
+        if not user_email or not isinstance(user_email, str):
+            raise ValueError("user must be a User object or a valid email")
+
+        # Check if the user exists in the organization
+        user_id = ApiKey._get_user(client, user_email)
+        if not user_id:
+            raise ValueError(
+                f"User with email '{user_email}' does not exist in the organization"
+            )
+
+        role_name = role.name if hasattr(role, "name") else role
+        if not role_name or not isinstance(role_name, str):
+            raise ValueError("role must be a Role object or a valid role name")
+
+        allowed_roles = ApiKey._get_available_api_key_roles(client)
+        if role_name not in allowed_roles:
+            raise ValueError(
+                f"Invalid role specified. Allowed roles are: {allowed_roles}"
+            )
+
+        validity_seconds = 0
+        if validity < 0:
+            raise ValueError("validity must be a positive integer")
+
+        if not isinstance(time_unit, TimeUnit):
+            raise ValueError("time_unit must be a valid TimeUnit enum value")
+
+        validity_seconds = validity * time_unit.value
+
+        if validity_seconds < TimeUnit.MINUTE.value:
+            raise ValueError("Minimum validity period is 1 minute")
+
+        max_seconds = 25 * TimeUnit.WEEK.value
+        if validity_seconds > max_seconds:
+            raise ValueError(
+                "Maximum validity period is 6 months (or 25 weeks)"
+            )
+
+        query_str = """
+         mutation CreateUserApiKeyPyApi($name: String!, $userEmail: String!, $role: String, $validitySeconds: Int) {
+             createApiKey(
+                 data: { name: $name, targetUserEmailId: $userEmail, role: $role, validitySeconds: $validitySeconds }
+             ) {
+                 id  
+                 jwt
+             }
+         }
+         """
+
+        params = {
+            "name": name,
+            "userEmail": user_email,
+            "role": role_name,
+            "validitySeconds": validity_seconds,
+        }
+
+        try:
+            result = client.execute(query_str, params)
+            api_key_result = result.get("createApiKey")
+
+            if not api_key_result:
+                raise LabelboxError(
+                    "Failed to create API key. No data returned from the server."
+                )
+
+            return api_key_result
+
+        except Exception as e:
+            raise LabelboxError(str(e)) from e
```

Comment:
i think `.execute()` would already throw error here if something is not right. So you could simplified this further by removing try/catch block.

Reply:
There are 2 things: 
- `execute()` didn't manage to create an API key and this justifies to raise an exception (execute wouldn't fail)
- `execute()`threw an exception and I want it to be propagated.